src/include/migraphx/instruction.hpp

```diff
@@ -96,6 +106,9 @@ struct instruction
     std::vector<instruction_ref> output;
     std::vector<instruction_ref> arguments;
     literal lit;
+
+    public:
+    int id = -1;
```

Comment:
This should be `std::size_t` that defaults to `0`. Also, an `id` parameter should be added to all of the constructors(and it should be set when inserting instructions).

Reply:
Having to add it to all constructors is not a good design.  If you use c++ class object, you don't need to do this.  I completely do not buy the typeless philosophy.  Nobody is using this.
Anyway, I will remove this id when everything else is cleared.

Reply:
Ignore my precious comment. I misunderstood.  
But "id" is meant to be a scratch field used by passes to track graph transformations.  This is a workaround since currently instructions do not have their own identify.

Scratch fields do not need to be maintained.

Reply:
> This is a workaround since currently instructions do not have their own identify.

I see. I understood this to be added so each instruction have their own identity. And the id is set by the program.cpp when the instruction is added.

Reply:
The pass might need to reassign values to id so that it is continuous and easy to track.